### DISCUSSION TITLE
Use standard attributes formatting with `f.inputs`

### DISCUSSION THREAD

**Discussion by amiel** [activeadmin/activeadmin#8436] [Ideas]
Currently (in both v4 and v3), html attributes passed to the `f.inputs` block in the `form` section of a resource, [are rendered](https://github.com/activeadmin/activeadmin/blob/v4.0.0.beta9/lib/active_admin/views/components/active_admin_form.rb#L135) using `html_options.map { |k, v| %Q{#{k}="#{v}"} }.join(" ")`.

This results in a different behavior from other places that take html attributes. For example:

```ruby
f.input :example, input_html: {data: {an_example: "testing"}}
# => <input ... data-an-example="testing"

f.inputs "A Group", data: {an_example: "testing"}
# => <fieldset data="{an_example=>"testing"}"
```

This can easily be resolved by using the Rails `tag.attributes` helper. For example, the following change is working well for me.

```diff
@@ -10,7 +10,7 @@ def build(_form_builder, *args, &)
         legend = args.shift if args.first.is_a?(::String)
         legend = html_options.delete(:name) if html_options.key?(:name)
         legend_tag = legend ? "<legend><span>#{ERB::Util.html_escape(legend)}</span></legend>" : ""
-        fieldset_attrs = html_options.map { |k, v| %(#{k}="#{v}") }.join(" ")
+        fieldset_attrs = tag.attributes html_options
 
         @opening_tag = "<fieldset #{fieldset_attrs}>#{legend_tag}<ol>"
         @closing_tag = "</ol></fieldset>"
```

As per the contributing guidelines, I'm opening this discussion to see if there's any interest in this fix (I'm happy to write up a PR).

I'm also wondering if there's any willingness to backport a fix like this to v3.

Thank you so much for your consideration!

**Comment by tagliala** [comment]
Refs: 
- #4518
- https://github.com/activeadmin/activeadmin/blob/06cac275bec0d74b282fa7606bc4deb5800db764/lib/active_admin/views/components/active_admin_form.rb#L135

Hi, please wait for a confirmation but I think that using `tag.attributes` is much better

**Comment by mgrunberg** [comment]
I like the idea of using `tag.attributes`. It would be good to use `tag.legend` to generate the `<legend>` too.

I did a quick test and using `tag.attributes` doesn't break #4518.

Just note that `tag.attributes` [was introduced in Rails 7.0](https://github.com/rails/rails/commit/7503ef9490583f7b2c3a475edd2bb1b7de0afcac) and we support Rails 6.1. I don't see another option than using it's implementation `tag.tag_options(html_options).to_s.strip.html_safe`

**Comment by amiel** [comment]
Here's a draft solution that incorporates suggestions from @mgrunberg: https://github.com/activeadmin/activeadmin/pull/8439

**Reply by javierjulio** [reply]
@amiel thank you again. The change looks good but we want to add a conditional so the older method is only used if running Rails 6.1, otherwise always use the current `tag.attributes` approach instead. Since we run the test suite for Rails 6.1 and 7.x, the newly added test should add coverage for both versions so don't anticipate any code coverage issues.

**Reply by amiel** [reply]
That makes sense. I'll do that now.

**Comment by amiel** [comment]
Thank you everyone for the input. Any continued discussion can happen in #8439.
